packages/api/src/aggregate/WhereClause.ts

```diff
@@ -195,5 +211,41 @@ export type WhereClause<
   | AndWhereClause<T>
   | NotWhereClause<T>
   | {
-    [P in PropertyKeys<T>]?: FilterFor<CompileTimeMetadata<T>["properties"][P]>;
+    [P in PropertyKeys<T>]?:
+      | RequireExactlyOne<
+        Pick<
+          WhereClauseKeys<T, P>,
+          NonNeverKeys<
+            WhereClauseKeys<T, P>
+          >
+        >
+      >
+      | FilterRawType<CompileTimeMetadata<T>["properties"][P]>;
```

Comment:
Use this so people can still pass in raw number, strings,booleans instead of having to do `$eq`